File: 백준/Silver/3273. 두 수의 합/두 수의 합.swift
import Foundation

let N = Int(readLine()!)!

var arr = readLine()!.split(separator: " ").map { Int($0)! }

let X = Int(readLine()!)!

var l = 0
var r = N - 1
var answer = 0
arr.sort()

while l < r {
    let sum = arr[l] + arr[r]
    if sum == X {
        answer += 1
        l += 1
        r -= 1
    } else if sum < X {
        l += 1
    } else {
        r -= 1
    }
}

print(answer)

File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
c = int(input())
arr = list(map(int, input().split()))
target = int(input())

arr.sort()

left = 0
right = len(arr) - 1
answer = 0

while left < right:
    sum = arr[left] + arr[right]
    
    if sum == target:
        answer += 1
        right -= 1
        left += 1
        
    elif sum > target:
        right -= 1
    else:
        left += 1
        
print(answer)